web/src/infra/common/components/base/FreeBoardGamesBar.tsx

```diff
@@ -1,55 +1,102 @@
+/* eslint-disable react/prop-types */
 import React from 'react';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
 import AppBar from '@material-ui/core/AppBar';
 import Toolbar from '@material-ui/core/Toolbar';
 import Typography from '@material-ui/core/Typography';
 import FbgLogo from './media/fbg_logo_white_48.png';
 import { Link } from 'infra/i18n';
 import { home } from 'infra/navigation';
+import { makeStyles } from '@material-ui/core';
+import { nextI18Next } from 'infra/i18n/config';
+import LanguageIcon from '@material-ui/icons/Language';
 
-interface FBGBarProps {
+type FBGBarProps = {
   FEATURE_FLAG_readyForDesktopView?: boolean;
   toolbarContent?: React.ReactNode;
-}
+};
+
+const handleLanguageChange = (event) => {
+  window.location.href = `/${event.target.value}`;
```

Comment:
I recommend using NextJS's router for this instead: https://nextjs.org/docs/api-reference/next/router#routerpush

Reply:
I tried that, but this wont work because the language files are loaded in the initial server render, so we need to force a full reload.